firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorViewController.swift

```diff
@@ -0,0 +1,294 @@
+    private var contentViewHeightConstraint: NSLayoutConstraint!
+    private let scaledRefreshButtonSize = UIFontMetrics.default.scaledValue(for: UX.passwordRefreshButtonHeight)
+    private let scaledHeaderImageSize = UIFontMetrics.default.scaledValue(for: UX.headerImageHeight)
+
+    // MARK: - Views
+
+    private lazy var contentView: UIView = .build { view in
+        view.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.content
+    }
+    private lazy var headerLabel: UILabel = .build { label in
+        label.numberOfLines = 0
+        label.font = FXFontStyles.Bold.body.scaledFont()
+        label.text = .PasswordGenerator.Title
+        label.isAccessibilityElement = true
+        label.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.headerLabel
+        label.accessibilityTraits = .header
+    }
+    private lazy var headerImageView: UIImageView = .build { imageView in
+        imageView.image = UIImage(named: StandardImageIdentifiers.Large.login)?.withRenderingMode(.alwaysTemplate)
+        imageView.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.headerImage
+    }
+    private lazy var descriptionLabel: UILabel = .build { label in
+        label.numberOfLines = 0
+        label.font = FXFontStyles.Regular.subheadline.scaledFont()
+        label.text = .PasswordGenerator.Description
+        label.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.descriptionLabel
+    }
+
+    private lazy var header: UIView = .build { view in
+        view.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.header
+    }
+
+    private lazy var passwordField: UIView = .build { field in
+        field.layer.borderWidth = UX.passwordFieldBorderWidth
+        field.layer.cornerRadius = UX.passwordFieldCornerRadius
+        field.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordField
+    }
+
+    private lazy var passwordLabel: UILabel = .build { label in
+        label.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordlabel
+    }
+
+    private lazy var passwordRefreshButton: UIButton = .build { button in
+        button.setImage(
+            UIImage(named: StandardImageIdentifiers.Large.arrowClockwise)?.withRenderingMode(.alwaysTemplate), for: .normal)
+        button.contentHorizontalAlignment = .fill
+        button.contentVerticalAlignment = .fill
+        button.accessibilityLabel = .PasswordGenerator.RefreshPasswordButtonA11yLabel
+        button.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordRefreshButton
+    }
+
+    private lazy var usePasswordButton: PrimaryRoundedButton = .build()
+
+    // MARK: - Initializers
+
+    init(windowUUID: WindowUUID,
+         themeManager: ThemeManager = AppContainer.shared.resolve(),
+         notificationCenter: NotificationProtocol = NotificationCenter.default,
+         currentTab: Tab) {
```

Comment:
curious, how would we be using the `currentTab` in this view? Do we need to pass in the Tab entirely or are there certain fields that we want to pull from it? I don't see it being used in the VC aside from initializing.

Reply:
Eventually in the middleware we'll need the webview and origin from from the tab when refreshing a generated password and for a few other middleware actions. 

In the case of refreshing a generated password, we use the webview of the tab to evaluate javascript code which generates a strong password for the site.

The password generation logic that I described above would be handled by the middleware, triggered by the action `PasswordGeneratorAction.userTappedRefreshPassword`. That action would be triggered on click of the refresh button within the` PasswordGeneratorViewController`. 

I could also inject the methods that require the tab into the `PasswordGeneratorViewController` if that makes more sense?

I'm happy to hop on a call if you'd like to discuss it further :)

Reply:
okay thank you for the clarification! sounds good, we can keep what you have here. Depending on the case, sometimes its easier to test and keep things decoupled if we pass in the string instead of the whole object that contains the string. But it seems we may need more than that for the pw work.